### PR TITLE
ITKDev - something like this

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to issue

https://jira.itkdev.dk/browse/CMN-371

#### Description

Something similar module

**will remove specific branches in composer file**

#### Screenshot of the result

image

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

If you have any further comments or questions for the reviewer them please add them here.

### CODE REVIEW

File: web/modules/custom/dpl_something_similar/src/Plugin/Block/SomethingSimilarBlock.php [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+   * @return mixed[]
+   *   The app render array.
+   */
+  public function build() {
+    $fbsConfig = $this->configFactory->get('dpl_fbs.settings');
+    $publizonConfig = $this->configFactory->get('dpl_publizon.settings');
+    $data = [
+      // Urls.
+      "fbs-base-url" => $fbsConfig->get('base_url'),
+      "publizon-base-url" => $publizonConfig->get('base_url'),
```
Comment (cableman) [inline web/modules/custom/dpl_something_similar/src/Plugin/Block/SomethingSimilarBlock.php:82]: These base url's are also added by `DplReactAppsController::externalApiBaseUrls()` in line 93

**Review by cableman — CHANGES_REQUESTED** [review]

File: web/modules/custom/dpl_something_similar/src/Plugin/Block/SomethingSimilarBlock.php (hunk 1 above)
Comment (sinejespersen) [reply web/modules/custom/dpl_something_similar/src/Plugin/Block/SomethingSimilarBlock.php]: removed

**Review by cableman — APPROVED** [review]

File: web/modules/custom/dpl_something_similar/src/Controller/DplSomethingSimilarController.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+   * @param \Drupal\Core\Block\BlockManagerInterface $blockManager
+   *   Drupal block manager.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   Drupal renderer service.
+   */
+  public function __construct(
+    private BlockManagerInterface $blockManager,
+    private RendererInterface $renderer
+  ) {
+  }
```
Comment (spaceo) [inline web/modules/custom/dpl_something_similar/src/Controller/DplSomethingSimilarController.php:28]: Is this intentionally empty if so then please document it

File: web/modules/custom/dpl_something_similar/src/Controller/DplSomethingSimilarController.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  public function list(): array {
+    $config = [];
+
+    /** @var \Drupal\dpl_something_similar\Plugin\Block\SomethingSimilarBlock $plugin_block */
+    $plugin_block = $this->blockManager->createInstance('dpl_something_similar_block', $config);
+
+    $access_result = $plugin_block->access($this->currentUser());
+    if (is_object($access_result) && $access_result->isForbidden() || is_bool($access_result) && !$access_result) {
+      throw new AccessDeniedHttpException();
+    }
```
Comment (spaceo) [inline web/modules/custom/dpl_something_similar/src/Controller/DplSomethingSimilarController.php:62]: Please see comment here:
https://github.com/danskernesdigitalebibliotek/dpl-cms/pull/297#discussion_r1192212982

File: web/modules/custom/dpl_something_similar/src/Plugin/Block/SomethingSimilarBlock.php [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      // Urls.
+      'dpl-cms-base-url' => DplReactAppsController::dplCmsBaseUrl(),
+      'material-url' => DplReactAppsController::materialUrl(),
+      'something-similar-title-text' => $this->t("Other materials", [], ['context' => 'Something similar']),
+      'material-by-author-text' => $this->t("By", [], ['context' => 'Something similar']),
+      'material-and-author-text' => $this->t("and", [], ['context' => 'Something similar']),
+      'something-similar-by-the-same-author-text' => $this->t("By the same author", [], ['context' => 'Something similar']),
+      'something-similar-something-similar-author-text' => $this->t("Something similar", [], ['context' => 'Something similar']),
+      'add-to-favorites-aria-label-text' => $this->t("Add element to favorites list", [], ['context' => 'Something similar (Aria)']),
+      'remove-from-favorites-aria-label-text' => $this->t("Remove element from favorites list", [], ['context' => 'Something similar (Aria)']),
```
Comment (spaceo) [inline web/modules/custom/dpl_something_similar/src/Plugin/Block/SomethingSimilarBlock.php:87]: Good idea to sort props....

**Review by spaceo — COMMENTED** [review]
I have some comments

File: web/modules/custom/dpl_something_similar/src/Controller/DplSomethingSimilarController.php (hunk 3 above)
Comment (cableman) [reply web/modules/custom/dpl_something_similar/src/Controller/DplSomethingSimilarController.php:62]: Ticket have been create to do this later on.

File: web/modules/custom/dpl_something_similar/src/Controller/DplSomethingSimilarController.php (hunk 2 above)
Comment (cableman) [reply web/modules/custom/dpl_something_similar/src/Controller/DplSomethingSimilarController.php:28]: Yes, the promoter pattern

File: web/modules/custom/dpl_something_similar/src/Plugin/Block/SomethingSimilarBlock.php (hunk 4 above)
Comment (cableman) [reply web/modules/custom/dpl_something_similar/src/Plugin/Block/SomethingSimilarBlock.php]: Done

**Comment by cableman** [comment]
Rebased on develop branch

**Review by cableman — APPROVED** [review]

File: web/modules/custom/dpl_something_similar/src/Controller/DplSomethingSimilarController.php (hunk 2 above)
Comment (spaceo) [reply web/modules/custom/dpl_something_similar/src/Controller/DplSomethingSimilarController.php:28]: Check

File: web/modules/custom/dpl_something_similar/src/Controller/DplSomethingSimilarController.php (hunk 3 above)
Comment (spaceo) [reply web/modules/custom/dpl_something_similar/src/Controller/DplSomethingSimilarController.php:62]: Splendid

**Review by spaceo — APPROVED** [review]
LGTM

**Pull request merged**